### PR TITLE
pkg/store/ratelimited: log offending key

### PR DESCRIPTION
**Pull request by squat**
This commit ensures that the key being ratelimited is logged along with
the error so admins can diagnose who is causing problems.

cc @brancz @aditya-konarde @s-urbaniak

### CODE REVIEW

**Comment by brancz** [comment]
cc @paulfantom that CI failure looks suspicious

**Comment by squat** [comment]
@paulfantom looks like there at some go module errors on the generate test. Can you take a look?

**Comment by s-urbaniak** [comment]
this seems to be some permission issue in the build container:
```
go: launchpad.net/gocheck@v0.0.0-20140225173054-000000000087: bzr branch --use-existing-dir https://launchpad.net/~niemeyer/gocheck/trunk . in /home/prow/go/pkg/mod/cache/vcs/f46ce2ae80d31f9b0a29099baa203e3b6d269dace4e5357a2cf74bd109e13339: exit status 4:
	failed to open trace file: [Errno 13] Permission denied: '/.bzr.log'
```

**Comment by s-urbaniak** [comment]
@squat just one small objection. We use this error to set an http response code:
https://github.com/openshift/telemeter/blob/f3fb4ee5181e266c7dc53a8139ae6495097a7754/pkg/http/server/server.go#L131-L132

With this change this won't work any more, and clients will most likely receive a 500 status code.

**Comment by s-urbaniak** [comment]
@squat just a suggestion. We could add another method to the newly added error type:
```
type ErrWriteLimitReached string

func (e ErrWriteLimitReached) Error() string {
	return fmt.Sprintf("write limit reached for key %q", e)
}

func (_ ErrWriteLimitReached) IsRateLimited() bool {
	return true
}
```

And then in the http server code we assert the above against an anonymous interface:
```
type rateLimitingErr interface {
	IsRateLimited() bool
}

if limiterr, ok := err.(rateLimitingErr); ok && limiterr.IsRateLimited() {
	http.Error(w, err.Error(), http.StatusTooManyRequests) 
} 
```

wdyt?

**Comment by squat** [comment]
@s-urbaniak, I had already modified that line so it would continue to work as it does today: https://github.com/openshift/telemeter/blob/607ee33eb09cc0a3179afa2282a1efe41ed2c5e2/pkg/http/server/server.go#L131

In fact, if I hadn’t updated it, the code wouldn’t compile at all.

**Comment by squat** [comment]
> this seems to be some permission issue in the build container:
> 
> ```
> go: launchpad.net/gocheck@v0.0.0-20140225173054-000000000087: bzr branch --use-existing-dir https://launchpad.net/~niemeyer/gocheck/trunk . in /home/prow/go/pkg/mod/cache/vcs/f46ce2ae80d31f9b0a29099baa203e3b6d269dace4e5357a2cf74bd109e13339: exit status 4:
> 	failed to open trace file: [Errno 13] Permission denied: '/.bzr.log'
> ```

Yes, this is what brancz and I were referencing earlier. It must be related to the recent CI and/or build-system changes that ocurre recently.

**Comment by paulfantom** [comment]
I think CI failure is similar to https://github.com/coreos/prometheus-operator/pull/2611 and solution is in https://github.com/coreos/prometheus-operator/pull/2611#issuecomment-494432406 :)

TL;DR We need python to download go libraries :man_shrugging:

**Comment by paulfantom** [comment]
Holding to stop unnecessary retests
/hold

**Comment by squat** [comment]
But why did the dependencies change at all? We never had an issue downloading go deps in this project so this is suggesting there has been some dependency creep. We should probably find and revert that change

**Comment by paulfantom** [comment]
We never had this problem because everything was done in travis where users are properly mapped.

Fix is in https://github.com/openshift/release/pull/3980

**Comment by paulfantom** [comment]
@squat you were right about dependencies. We migrated to go 1.12, so every execution of `go get ...` adds new dependency to `go.mod` and we are doing `go get` in Makefile  to download some binaries. Those instructions should be prefixed with `GO111MODULE=off` or removed completely.

**Comment by paulfantom** [comment]
#181 landed and this needs rebase now.

**Comment by squat** [comment]
thanks for the heads up @paulfantom, updated
